pkg/republisher/uuid_republisher_test.go

```diff
@@ -198,6 +198,107 @@ func TestFoundInNoneAndNotFoundInDocStore_Ok(t *testing.T) {
 	assert.True(t, strings.HasSuffix(errs[0].Error(), "wasn't found in any of the native-store's collections and it's not an ImageSet"))
 }
 
+func TestUUIDInUCAndMethodeRepublishedInUC(t *testing.T) {
+	mockedUCRepublisher := new(mockUCRepublisher)
+	mockedDocStoreClient := new(mockDocStoreClient)
+	msg := OKMsg{
+		uuid:                     "64bc4319-cd22-43e9-8b12-358622d7a5ba",
+		tid:                      "prefix1tid_123",
+		collectionName:           "universal-content",
+		collectionOriginSystemID: "cct",
+		sizeBytes:                1024,
+		notifierAppName:          "cms-notifier",
+	}
+
+	msg2 := OKMsg{
+		uuid:                     "b3ec9282-1073-46ad-9d44-144dad7fe956",
+		tid:                      "prefix1",
+		collectionName:           "pac-metadata",
+		collectionOriginSystemID: "http://cmdb.ft.com/systems/pac",
+		sizeBytes:                1024,
+		notifierAppName:          "cmsMetadataNotifier",
+	}
+
+	var nilMsg *OKMsg
+	mockedUCRepublisher.On("RepublishUUIDFromCollection", "64bc4319-cd22-43e9-8b12-358622d7a5ba", mock.MatchedBy(func(tid string) bool { return strings.HasPrefix(tid, "prefix1") }), testCollections["methode"]).Return(nilMsg, false, nil)
+	mockedUCRepublisher.On("RepublishUUIDFromCollection", "64bc4319-cd22-43e9-8b12-358622d7a5ba", mock.MatchedBy(func(tid string) bool { return strings.HasPrefix(tid, "prefix1") }), testCollections["universal-content"]).Return(&msg, true, nil)
+	mockedUCRepublisher.On("RepublishUUIDFromCollection", "64bc4319-cd22-43e9-8b12-358622d7a5ba", mock.MatchedBy(func(tid string) bool { return strings.HasPrefix(tid, "prefix1") }), testCollections["v1-metadata"]).Return(nilMsg, false, nil)
+	mockedUCRepublisher.On("RepublishUUIDFromCollection", "64bc4319-cd22-43e9-8b12-358622d7a5ba", mock.MatchedBy(func(tid string) bool { return strings.HasPrefix(tid, "prefix1") }), testCollections["pac-metadata"]).Return(&msg2, true, nil)
+	mockedUCRepublisher.On("RepublishUUIDFromCollection", "64bc4319-cd22-43e9-8b12-358622d7a5ba", mock.MatchedBy(func(tid string) bool { return strings.HasPrefix(tid, "prefix1") }), testCollections["wordpress"]).Return(nilMsg, false, nil)
+	mockedUCRepublisher.On("RepublishUUIDFromCollection", "64bc4319-cd22-43e9-8b12-358622d7a5ba", mock.MatchedBy(func(tid string) bool { return strings.HasPrefix(tid, "prefix1") }), testCollections["video"]).Return(nilMsg, false, nil)
+	r := NewNotifyingUUIDRepublisher(mockedUCRepublisher, mockedDocStoreClient, testCollections)
+
+	msgs, errs := r.Republish("64bc4319-cd22-43e9-8b12-358622d7a5ba", "prefix1", ScopeBoth)
+
+	assert.Equal(t, 2, len(msgs))
```

Comment:
How is this test checking which collection is being returned between `methode` and `universal-content`? It only counts the # of returned messages.

Reply:
it does, not it only assures that there're exactly 2 msgs, the range statement checks the value of those msgs.

Reply:
Thanks for pointing that out.